### PR TITLE
add test for getOnePost

### PR DESCRIPTION
**Pull request by kevinchu93**
first test, the test seems redundant to me.
@ericchu94

### CODE REVIEW

File: test/postServiceTest.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            sut.getOnePost(5);
+            assert.equal(sut.getOnePost(5), 5);
+            stub.restore()
+        });
+    });
+});
+    
+
+
+
```
Comment (ericchu94) [inline test/postServiceTest.js]: Extra blank lines

File: test/postServiceTest.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+postService = new PostService;
+
+
+describe('PostService', () => {
+    describe('getOnePost(post_id)', () => {
+        it('should return correct post', () => {
+            const stub = sinon.stub(Post, 'find').returns(5);
+            const sut = new PostService;
+            sut.getOnePost(5);
+            assert.equal(sut.getOnePost(5), 5);
```
Comment (ericchu94) [inline test/postServiceTest.js]: You should only call `getOnePost()` once in this test, and save its return value

File: test/postServiceTest.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const mocha = require('mocha');
+const sinon = require('sinon');
+const assert = require('assert');
+const Post = require('../models/Post');
+const mongoose = require('../mongoose');
+
+const PostService = require('../services/PostService');
+
+postService = new PostService;
```
Comment (ericchu94) [inline test/postServiceTest.js]: This declares a global variable `postService`, and also is not how you create new instances.

should be `const postService = new PostService();`

Comment (ericchu94) [inline test/postServiceTest.js]: `postService` is unused

File: test/postServiceTest.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+
+describe('PostService', () => {
+    describe('getOnePost(post_id)', () => {
+        it('should return correct post', () => {
+            const stub = sinon.stub(Post, 'find').returns(5);
+            const sut = new PostService;
```
Comment (ericchu94) [inline test/postServiceTest.js]: `new PostService()`

File: test/postServiceTest.js (hunk 2 above)
Comment (ericchu94) [reply test/postServiceTest.js]: This test will pass if `getOnePost` is defined as follows:
```
getOnePost(id) {
    return id;
}
```
or
```
getOnePost(id) {
    return 5;
}
```

File: test/postServiceTest.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+            sut.getOnePost(5);
+            assert.equal(sut.getOnePost(5), 5);
+            stub.restore()
```
Comment (ericchu94) [inline test/postServiceTest.js]: semicolon

**Review by ericchu94 — CHANGES_REQUESTED** [review]

File: test/postServiceTest.js (hunk 5 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: fixed

File: test/postServiceTest.js (hunk 1 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: fixed

File: test/postServiceTest.js (hunk 3 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: fixed

Comment (kevinchu93) [reply test/postServiceTest.js]: fixed

File: test/postServiceTest.js (hunk 4 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: fixed

File: test/postServiceTest.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('PostService', () => {
+    describe('getOnePost(post_id)', () => {
+        it('should return correct post', () => {
+            const stub = sinon.stub(Post, 'find').returns(5);
+            const sut = new PostService();
+            post_id = sut.getOnePost(5);
+            assert.equal(post_id, 5);
+            stub.restore();
+        });
```
Comment (ericchu94) [inline test/postServiceTest.js]: There are a couple things you should be testing in this describe section.
1. The function you mocked should be called with the right arguments
2. The return value should be the value returned via the stub

File: test/postServiceTest.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const mocha = require('mocha');
+const sinon = require('sinon');
+const assert = require('assert');
+const Post = require('../models/Post');
+const mongoose = require('../mongoose');
```
Comment (ericchu94) [inline test/postServiceTest.js]: unused variable. Please run eslint before you submit a review

**Review by ericchu94 — CHANGES_REQUESTED** [review]

File: test/postServiceTest.js (hunk 7 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: fixed

File: test/postServiceTest.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+const Post = require('../models/Post');
+
+const PostService = require('../services/PostService');
+
+describe('PostService', () => {
+  describe('getOnePost(post_id)', () => {
+    it('should return correct post', () => {
+      const stub = sinon.stub(Post, 'find').returns(5);
+      const sut = new PostService();
+      const postId = sut.getOnePost(5);
```
Comment (ericchu94) [inline test/postServiceTest.js]: `getOnePost` returns a post, not a post id

File: test/postServiceTest.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+const sinon = require('sinon');
+const assert = require('assert');
+const Post = require('../models/Post');
+
+const PostService = require('../services/PostService');
+
+describe('PostService', () => {
+  describe('getOnePost(post_id)', () => {
+    it('should return correct post', () => {
+      const stub = sinon.stub(Post, 'find').returns(5);
```
Comment (ericchu94) [inline test/postServiceTest.js]: If you stub out a function you should assert that it is called with the right params

File: test/postServiceTest.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+      const sut = new PostService();
+      const postId = sut.getOnePost(5);
+      assert.equal(postId, 5);
+      stub.restore();
+    });
```
Comment (ericchu94) [inline test/postServiceTest.js]: This test still isn't sufficient for a minimal test

**Review by ericchu94 — CHANGES_REQUESTED** [review]

File: test/postServiceTest.js (hunk 8 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: changed

File: test/postServiceTest.js (hunk 9 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: added assertion

File: test/postServiceTest.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    it('should be called with correct parameters', () => {
+      const stub = sinon.stub(Post, 'findById');
+      PostService.getOnePost(5);
+      sinon.assert.calledWith(stub, 5);
+      stub.restore();
+    });
+  });
+  describe('getAllPosts()', () => {
```
Comment (ericchu94) [inline test/postServiceTest.js]: Remove this test; this test is out of scope for this PR. It's also bad practice to change the scope of the PR

File: test/postServiceTest.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    it('should be called with no parameters', () => {
+      const stub = sinon.stub(Post, 'find');
+      PostService.getAllPosts();
+      sinon.assert.calledWith(stub);
+      stub.restore();
+    });
+  });
+  describe('createNewPost(title, author, body)', () => {
```
Comment (ericchu94) [inline test/postServiceTest.js]: Remove this test; this test is out of scope for this PR. It's also bad practice to change the scope of the PR

File: test/postServiceTest.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+const PostService = require('../services/PostService');
+
+describe('PostService', () => {
+  describe('getOnePost(post_id)', () => {
+    it('should return correct post', () => {
+      const stub = sinon.stub(Post, 'findById').returns(5);
+      const post = PostService.getOnePost(5);
+      assert.equal(post, 5);
+      stub.restore();
+    });
```
Comment (ericchu94) [inline test/postServiceTest.js]: The problem with stubbing and calling your method like this is that the following implementation will pass your tests:
```
getOnePost(postId) {
    Post.findById(postId);
    return postId;
}
```

Your assertion that the return value of `findById()` is the same as `getOnePost()`  is too weak

File: test/postServiceTest.js [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    it('should be called with correct parameters', () => {
```
Comment (ericchu94) [inline test/postServiceTest.js]: kind of misleading since "it" refers to `getOnePost()`

**Review by ericchu94 — CHANGES_REQUESTED** [review]

File: test/postServiceTest.js (hunk 13 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: fixed by having different values for getOnePost parameter and findById parameter

File: test/postServiceTest.js (hunk 11 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: removed

File: test/postServiceTest.js (hunk 12 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: removed

File: test/postServiceTest.js (hunk 14 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: modified statement within "it"

File: test/postServiceTest.js [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+const sinon = require('sinon');
+const assert = require('assert');
+const Post = require('../models/Post');
+const PostService = require('../services/PostService');
+
+describe('PostService', () => {
+  describe('getOnePost(post_id)', () => {
+    it('should return correct post', () => {
+      const stub = sinon.stub(Post, 'findById').returns('post');
+      const post = PostService.getOnePost('postId');
```
Comment (ericchu94) [inline test/postServiceTest.js]: This is really weird since this unit test basically says that `getOnePost()` accepts a String as a parameter

**Review by ericchu94 — CHANGES_REQUESTED** [review]

File: test/postServiceTest.js (hunk 15 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: changed

**Review by ericchu94 — APPROVED** [review]
